components/movementsensor/gpsnmea/pmtkI2C.go

```diff
@@ -192,21 +192,29 @@ func (g *PmtkI2CNMEAMovementSensor) GetBusAddr() (board.I2C, byte) {
 func (g *PmtkI2CNMEAMovementSensor) Position(ctx context.Context, extra map[string]interface{}) (*geo.Point, float64, error) {
 	g.mu.RLock()
 	defer g.mu.RUnlock()
-	return g.data.location, g.data.alt, g.lastError
+	// Return the most recent error, and clear it for next time.
+	lastError := g.lastError
+	g.lastError = nil
```

Comment:
[minor] maybe a function that does this:
```
func (g *movementSensor) returnAndResetLastError() error {
      g.errMu.RLock()
      defer g.errMu.RUnlock()
      errOut := g.lastErr
      g.lastErr = nil
      return errOut
}
```

`setLastError` and `getLastError` in gpsnmea, cameramono have similar logic. maybe there's a way you could edit those functions to do this instead? it seems like a lot of overlap.

Reply:
> maybe there's a way you could edit those functions to do this instead?

Thanks to @gvaradarajan for this suggestion: we should make a `LastError` object that has `Get()` and `Set(error)` functions on it, and then each paragraph I added and the corresponding parts of the other files you mentioned all become a 1-liners. Not sure if I'll get to this today, but I really like that design and will do it sometime soon.